### PR TITLE
[23929] Add tests for empty dispose messages

### PR DESCRIPTION
**Pull request by emiliocuestaf**
This PR adds a test to check that dispose messages with empty payload are correctly forwarded in the DDS Pipe/Router ecosystem.

This PR depends on https://github.com/eProsima/DDS-Pipe/pull/172 and must be merged after it!

CI results with those changes in DDSPipe: https://github.com/eProsima/DDS-Router/actions/runs/21034096726

### CODE REVIEW

File: ddsrouter_core/test/blackbox/ddsrouter_core/dds/local/CMakeLists.txt [hunk 1]
```diff
@@ -89,3 +96,4 @@ add_blackbox_executable(
     "${TEST_LIST}"
     "${TEST_NEEDED_SOURCES}"
     "${TEST_EXTRA_HEADERS}")
+
```
Comment (raulojeda22) [inline ddsrouter_core/test/blackbox/ddsrouter_core/dds/local/CMakeLists.txt]: Extra endline

File: ddsrouter_core/test/blackbox/ddsrouter_core/dds/local/DDSTestLocalInstances.cpp [hunk 2]
```diff
@@ -0,0 +1,823 @@
...
+        auto part = std::make_shared<participants::SimpleParticipantConfiguration>();
+        part->id = core::types::ParticipantId("participant_1");
+        part->domain.domain_id = 1u;
+        conf.participants_configurations.insert({types::ParticipantKind::simple, part});
+    }
+
+    return conf;
+}
+
+DdsRouterConfiguration dds_test_dds_participants_config()
```
Comment (raulojeda22) [inline ddsrouter_core/test/blackbox/ddsrouter_core/dds/local/DDSTestLocalInstances.cpp:87]: Do you need the XML profiles here to test a specific thing about that? Add a comment header here explaining it.

File: ddsrouter_core/test/blackbox/ddsrouter_core/dds/local/DDSTestLocalInstances.cpp [hunk 3]
```diff
@@ -0,0 +1,823 @@
...
+
+    {
+        auto part = std::make_shared<participants::XmlParticipantConfiguration>();
+        part->id = core::types::ParticipantId("participant_1");
+        part->participant_profile.set_value("participant_domain_1");
+        conf.participants_configurations.insert({types::ParticipantKind::xml, part});
+    }
+
+    return conf;
+}
```
Comment (raulojeda22) [inline ddsrouter_core/test/blackbox/ddsrouter_core/dds/local/DDSTestLocalInstances.cpp:135]: Just one more blank line for consistency

File: ddsrouter_core/test/blackbox/ddsrouter_core/dds/local/DDSTestLocalInstances.cpp [hunk 4]
```diff
@@ -0,0 +1,823 @@
...
+        {
+            std::this_thread::sleep_for(std::chrono::milliseconds(time_between_samples));
+        }
+    }
+    // All samples received, now unregister key from publisher and check that subscriber has received
+    // a not alive no writers event
+    ASSERT_TRUE(publisher.unregister_key(msg) == fastdds::dds::RETCODE_OK);
+    std::this_thread::sleep_for(std::chrono::milliseconds(DEFAULT_SUBSCRIBER_MESSAGE_RECEPTION));
+
+    // TODO: Add comment here
```
Comment (raulojeda22) [inline ddsrouter_core/test/blackbox/ddsrouter_core/dds/local/DDSTestLocalInstances.cpp]: Add the comment

File: ddsrouter_core/test/blackbox/ddsrouter_core/dds/local/DDSTestLocalInstances.cpp [hunk 5]
```diff
@@ -0,0 +1,823 @@
...
```
Comment (raulojeda22) [inline ddsrouter_core/test/blackbox/ddsrouter_core/dds/local/DDSTestLocalInstances.cpp]: Add the comment

**Review by raulojeda22 — CHANGES_REQUESTED** [review]

**Review by raulojeda22 — DISMISSED** [review]
LGTM, CI tests and devutils pending
